apps/website/src/hooks/pixels.tsx

```diff
@@ -11,24 +22,377 @@ export const PIXEL_GRID_WIDTH = 200;
+    };
+  }
+
+  private stopDemo() {
+    if (this.updateInterval) {
+      clearInterval(this.updateInterval);
+      this.updateInterval = undefined;
+    }
+  }
+
+  private startDemo() {
+    this.updateInterval = setInterval(() => {
+      const donationId = `demo_donation_${Math.floor(Math.random() * 1000000)}`;
+      const identifier = `@demo_user_${Math.floor(Math.random() * 1000)}`;
+
+      const numberOfPixels = Math.floor(Math.random() * 2) + 1;
+
+      const fakeEvent: UpdateEvent = {
+        type: "update",
+        payload: {
+          identifier,
+          amount: numberOfPixels * 10000 + Math.floor(Math.random() * 3000),
+          pixels: Array.from({ length: numberOfPixels }).map(() => {
+            const color = getDemoColor();
+            return {
+              id: `pixel_${Math.floor(Math.random() * 1000000)}`,
+              donationId,
+              muralId: this.muralId,
+              email: null,
+              receivedAt: new Date(),
+              data: btoa(
+                String.fromCharCode(
+                  ...Array.from({ length: PIXEL_SIZE * PIXEL_SIZE }).flatMap(
+                    () => getShiftedColor(color),
+                  ),
+                ),
+              ),
+              identifier,
+              column: Math.floor(Math.random() * PIXEL_GRID_WIDTH),
+              row: Math.floor(Math.random() * PIXEL_GRID_HEIGHT),
+            };
+          }),
+        },
+      };
+
+      this.context = [...this.context, ...fakeEvent.payload.pixels];
+      this.callbacks.forEach((callback) => {
+        if (callback.type === "event") {
+          callback.callback(fakeEvent.payload);
+        }
+        if (callback.type === "update") {
+          callback.callback(this.context);
+        }
+      });
+    }, 1000);
+  }
+
+  public close() {
+    this.stopDemo();
+  }
+
+  public reconnect() {
+    this.stopDemo();
+    this.startDemo();
+  }
+}
+
+const PixelSyncProviderContext = createContext<
+  IPixelSyncProvider | IPixelStaticProvider | null
+>(null);
+
+export const PixelProvider = ({
+  children,
+  muralId,
+  type,
+}: {
+  muralId: string;
+  type: "sync" | "static";
+  children: ReactNode;
+}) => {
+  const sync = useMemo(() => {
+    // If the mural is static, we don't need to connect to the websocket
+    if (type === "static") {
+      return {
+        type: "static",
+        key: ["pixels", "static", muralId],
+        muralId,
+      } satisfies IPixelStaticProvider;
+    }
+
+    if (env.NEXT_PUBLIC_DONATIONS_DEMO_MODE || typeof window === "undefined") {
+      return new DemoPixelSyncProvider({ muralId });
+    }
+
+    if (!env.NEXT_PUBLIC_DONATIONS_MANAGER_URL) {
+      throw new Error("NEXT_PUBLIC_DONATIONS_MANAGER_URL is not set");
+    }
+
+    // If the mural is sync, we need to connect to the websocket
+    return new PixelSyncProvider({
+      url: `${env.NEXT_PUBLIC_DONATIONS_MANAGER_URL}/pixels/sync`,
+      muralId,
+    });
```

Comment:
Should `PixelSyncProvider` take a `muralId`, that feels misleading to me as it'll only ever return data for whichever mural it considers to be the live one?